### PR TITLE
Add ARM64 support to ilasm/ildasm

### PR DESCRIPTION
**Pull request by BruceForstall**
Remove IA-64/Itanium support.

Fixes https://github.com/dotnet/coreclr/issues/8084

### CODE REVIEW

**Comment by BruceForstall** [comment]
Not sure what to do about the ExportStubARM64Template -- whether it needs to be fixed or not. Currently, it's just a copy of the ARM32 one, with an error message if you hit that path.

The command line processing is a bit questionable, also: all parameters are processed by only looking at the first 3 characters (!) which isn't good enough to distinguish "ARM" and "ARM64", so for that case only I made it look at 5 characters. An alternative would be to use "A64" instead of "ARM64", but that seems ugly, too.

**Comment by BruceForstall** [comment]
@jkotas @RussKeldorph PTAL

**Comment by AntonLapounov** [comment]
https://github.com/dotnet/coreclr/blob/030a3ea9b8dbeae89c90d34441d4d9a1cf4a7de6/src/ilasm/main.cpp#L590
Is this option not supported for the ARM/ARM64 platform?  Just wanted to make sure.

File: src/ilasm/main.cpp [hunk 1]
```diff
@@ -188,17 +188,16 @@ extern "C" int _cdecl wmain(int argc, __in WCHAR **argv)
...
       printf("\n/32BITPREFERRED Create a 32BitPreferred image (PE32)");
       printf("\n/ENC=<file>     Create Edit-and-Continue deltas from specified source file");
       
       printf("\n\nKey may be '-' or '/'\nOptions are recognized by first 3 characters\nDefault source file extension is .il\n");
 
       printf("\nTarget defaults:");
-      printf("\n/PE64      => /PE64 /ITANIUM");
-      printf("\n/ITANIUM   => /PE64 /ITANIUM");
+      printf("\n/PE64      => /PE64 /X64");
       printf("\n/X64       => /PE64 /X64");
```
Comment (AntonLapounov) [inline src/ilasm/main.cpp]: Should we also add `/ARM64       => /PE64 /ARM64`?

Comment (BruceForstall) [reply src/ilasm/main.cpp]: Done.

**Comment by BruceForstall** [comment]
>  fprintf(stderr,"/32BITPREFERRED valid only with PE32/X86/ILONLY images\n"); 
> 
> Is this option not supported for the ARM/ARM64 platform? Just wanted to make sure.

I don't know. Seems unlikely?

**Comment by BruceForstall** [comment]
> ARM64 stub is not needed. The ARM is not actually needed either - you can delete it if you would like.

I removed the ARM64 one. I left the ARM one; that can be removed separately, if desired.

File: src/ilasm/main.cpp [hunk 2]
```diff
@@ -188,18 +188,18 @@ extern "C" int _cdecl wmain(int argc, __in WCHAR **argv)
       printf("\n/HIGHENTROPYVA  Set High Entropy Virtual Address capable PE32+ images (default for /APPCONTAINER)");
       printf("\n/NOCORSTUB      Suppress generation of CORExeMain stub");
       printf("\n/STRIPRELOC     Indicate that no base relocations are needed");
-      printf("\n/ITANIUM        Target processor: Intel Itanium");
       printf("\n/X64            Target processor: 64bit AMD processor");
```
Comment (AntonLapounov) [inline src/ilasm/main.cpp]: Nit: It makes sense to replace "64bit AMD" with just "x64".

Comment (BruceForstall) [reply src/ilasm/main.cpp]: @AntonLapounov I agree this makes sense, but I want to scope this change to only ARM64 work, and not add general improvement/cleanup.

**Comment by BruceForstall** [comment]
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked Innerloop Build and Test
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked no_tiered_compilation_innerloop Build and Test

**Comment by xiangzhai** [comment]
:mips-interest

**Pull request merged**